presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1147,13 +1299,14 @@ private void commitShared()
             committer.executeAddTableOperations();
             committer.executeAlterTableOperations();
             committer.executeAlterPartitionOperations();
-            committer.executeAddPartitionOperations();
-            committer.executeUpdateStatisticsOperations();
+            committer.executeAddPartitionOperations(transaction);
+            committer.executeUpdateStatisticsOperations(transaction);
         }
         catch (Throwable t) {
+            log.error(t, "Metastore commit failed - - rolling back");
```

Comment:
We shouldn't log the full exception here since we throw it below. But we could do it like
```java
log.warn("Rolling back due to metastore commit failure: %s", t.getMessage());

Reply:
Changed.